src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -378,9 +379,39 @@ void IdealGraphPrinter::visit_node(Node *n, bool edges, VectorSet* temp_set) {
         print_prop("block", C->cfg()->get_block(0)->_pre_order);
       } else {
         print_prop("block", block->_pre_order);
+        // Print estimated execution frequency, normalized within a [0,1] range.
+        buffer[0] = 0;
+        stringStream freq(buffer, sizeof(buffer) - 1);
+        // Higher precision has no practical effect in visualizations.
+        freq.print("%.8f", block->_freq / max_freq);
+        assert(freq.size() < sizeof(buffer), "size in range");
+        // Enforce dots as decimal separators, as required by IGV.
+        StringUtils::replace_no_expand(buffer, ",", ".");
+        print_prop("frequency", buffer);
       }
     }
 
+    switch (t->category()) {
```

Comment:
Might be more readable if switch cases are indented.

Reply:
Done.